subprojects/signing/src/main/java/org/gradle/plugins/signing/SigningExtension.java

```diff
@@ -305,7 +311,75 @@ public void execute(Sign task) {
         return result;
     }
 
-    private Sign createSignTaskFor(final CharSequence name, Action<Sign> taskConfiguration) {
+    /**
+     * Creates signing tasks that sign all publishable artifacts of the given publications.
+     *
+     * <p>The created tasks will be named "sign<i>&lt;publication name capitalized&gt;</i>Publication".
+     * That is, given a publication with the name "mavenJava" the created task will be named "signMavenJavaPublication".
+     *
+     * The signature artifacts for the created tasks are added to the publishable artifacts of the given publications.
+     *
+     * @param publications The publications whose artifacts are to be signed
+     * @return the created tasks.
+     * @since 4.8
+     */
+    @Incubating
+    public List<Sign> sign(Publication... publications) {
+        List<Sign> result = new ArrayList<Sign>(publications.length);
+        for (final Publication publication : publications) {
+            result.add(createSignTaskFor((PublicationInternal<?>) publication));
+        }
+        return result;
+    }
+
+    /**
+     * Creates signing tasks that sign all publishable artifacts of the given publication collection.
+     *
+     * <p>The created tasks will be named "sign<i>&lt;publication name capitalized&gt;</i>Publication".
+     * That is, given a publication with the name "mavenJava" the created task will be named "signMavenJavaPublication".
+     *
+     * The signature artifacts for the created tasks are added to the publishable artifacts of the given publications.
+     *
+     * @param publications The collection of publications whose artifacts are to be signed
+     * @return the created tasks.
+     * @since 4.8
+     */
+    @Incubating
+    public List<Sign> sign(DomainObjectCollection<Publication> publications) {
+        final List<Sign> result = new ArrayList<Sign>();
+        publications.all(new Action<Publication>() {
```

Comment:
I wasn't sure that was a valid use case... but since we support it in these other places, I guess we should do so here, too. So, I will add it. And it's so much fun... 😉

Reply:
Done in fecb2fcc804589f451e54fd1f5b3a0a09d4db660.